agent/src/android/root.ts

```diff
@@ -100,6 +100,264 @@ export namespace root {
     });
   };
 
+  const bypassRootBeer_isRooted = (success: boolean, ident: string): any => {
+    return wrapJavaPerform(() => {
+      const RootBeer = Java.use("com.scottyab.rootbeer.RootBeer");      
+      RootBeer.isRooted.overload().implementation = function() {
+        if (success) {
+            send(
+              c.blackBright(`[${ident}] `) +
+              `RootBeer->isRooted() check detected, marking as ${c.green("true")}.`,
+            );
+            return true;
+        } else {
+            send(
+              c.blackBright(`[${ident}] `) +
+              `RootBeer->isRooted() check detected, marking as ${c.green("false")}.`,
+            );
+            return false;
+        }
+        // call the original method
+        return this.isRooted.call(this);
+      };
+      return RootBeer;
+    });
+  }; 
+
+  const bypassRootBeer_checkForBinary = (success: boolean, ident: string): any => {
+    return wrapJavaPerform(() => {
+      const RootBeer = Java.use("com.scottyab.rootbeer.RootBeer");  
+      RootBeer.checkForBinary.overload('java.lang.String').implementation = function() {
+        if (success) {
+            send(
+              c.blackBright(`[${ident}] `) +
+              `RootBeer->checkForBinary() check detected, marking as ${c.green("true")}.`,
+            );
+            return true;
+        } else {
+            send(
+              c.blackBright(`[${ident}] `) +
+              `RootBeer->checkForBinary() check detected, marking as ${c.green("false")}.`,
+            );
+            return false;
+        }
+        // call the original method
+        return this.checkForBinary.overload('java.lang.String').call(this);
```

Comment:
In all of the added functions, the line that calls the original method is unreachable. In this case, the check for the value of `success` will flow into two blocks that will return, never reaching this line.